src/App.tsx

```diff
@@ -90,7 +90,7 @@ function Layout() {
         }
         if (
           user &&
-          (!user.id || user.changed) &&
+          (user?.loginState === "needs-registration" || user?.loginState === "needs-reregistration") &&
```

Comment:
```suggestion
          user.loginState === LoginState.NeedsRegistration || user.loginState === LoginState.NeedsReregistration
```

You already checked that user is set, so you don't need `user?.`. Regarding the enum see below.

Reply:
Done